tests/unit/test_utils.py

```diff
@@ -1336,16 +1336,38 @@ def test_deprecated_message_reads_well():
     )
 
 
+@pytest.mark.parametrize("global_options", [
+    None,
+    [],
+    ["--some", "--option"]
+])
+@pytest.mark.parametrize("no_user_cfg", [False, True])
 @pytest.mark.parametrize("unbuffered_output", [False, True])
```

Comment:
Each of these should be its own test IMO. There can be

* `test_make_setuptools_shim_args` (unparametrized, just testing `args[-1]` in more detail)
* `test_make_setuptools_shim_args__global_options()`
* `test_make_setuptools_shim_args__no_user_config()`
* `test_make_setuptools_shim_args__unbuffered_output()`

The reason is that it's not so elegant to have a parametrized test and then use if blocks inside to detect if you're in that test. Also, calling `@pytest.mark.parametrize()` in that way is making a multi-dimensional matrix of test cases which is overboard for something so simple. You only need to test the simple (one-dimensional) cases.

Reply:
I had the dilemma as to split the test case or parameterize, will simplify the test case.